tools/bash/clone_herbert_branch.sh

```diff
@@ -25,15 +25,21 @@ if [ "${herbert_branch}" = "" ]; then
 fi
 
 if [[ -d "${HERBERT_DIR}" ]]; then
-    echo_and_run "git -C ${HERBERT_DIR} fetch -all --tags" &&
-    echo_and_run "git -C ${HERBERT_DIR} reset --hard \"${herbert_branch}\""
+    pushd "${HERBERT_DIR}" || exit 1
+    echo_and_run "git fetch -all --tags" &&
+    echo_and_run "git reset --hard \"${herbert_branch}\""
+    popd || exit 1
+
 else
     git_clone_cmd="git clone ${HERBERT_URL} --depth 1"
     git_clone_cmd+=" --branch \"${herbert_branch}\" ${HERBERT_DIR}"
     echo_and_run "${git_clone_cmd}"
 fi
 
-echo -e "\nBuilding Herbert revision $(git -C ${HERBERT_DIR} rev-parse HEAD)..."
+pushd "${HERBERT_DIR}" || exit 1
```

Comment:
Do we need an equivalent to the powershell `Invoke-and-run` in the bash script now to remove repeats of the pushd/action/popd pattern?

If the action between the pushd and popd fails we should fail to the original directory.

Reply:
Did you mean `Invoke-In-Dir`? There's no bash equivalent of `try {} finally {}` as far as I know. Will `pushd; <code> || popd; popd` work? 

These are developer scripts, so I'm not sure if it's worth doing or not. This script is only used in build pipelines, so leaving us in a separate directory is not the end of the world - the subsequent `sh '...'` commands spawn new shell sessions from the workspace root anyway

Reply:
I did.

The problem with it leaving you in the wrong directory is more of an issue when running a local build (this used to happen regularly). 

Maybe `pushd && action && popd || popd` or simply `pushd && action; popd`